zstash/hpss_new.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import print_function, absolute_import
+
+import os.path
+import subprocess
+from .settings import logger
+
+
+def run_command(command, error_str):
+    completed_process = subprocess.run(command.split(), capture_output=True, encoding='utf-8')
+    status = completed_process.return_code
+    if status != 0:
+        logger.error(error_str)
+        logger.debug('stdout:\n%s', completed_process.stdout)
+        logger.debug('stderr:\n%s', completed_process.stderr)
+        raise Exception
+
+
+def hpss_transfer(hpss, file_path, put, keep=None):
+    if hpss == 'none':
+        return
+    if put:
+        transfer_word = 'to'
+        transfer_command = 'put'
+    else:
+        transfer_word = 'from'
+        transfer_command = 'get'
+    logger.info('Transferring file {} HPSS: {}'.format(transfer_word, file_path))
+    path, name = os.path.split(file_path)
+
+    # Need to be in local directory for hsi put to work
+    cwd = os.getcwd()
+    if path != '':
+        os.chdir(path)
+
+    # Transfer file using hsi put
+    command = 'hsi -q "cd {}; {} {}"'.format(hpss, transfer_command, name)
+    error_str = 'Transferring file {} HPSS: {}'.format(transfer_word, name)
+    run_command(command, error_str)
+
+    # Back to original working directory
+    if path != '':
+        os.chdir(cwd)
+
+    if put:
+        # Remove local file if requested
+        if not keep:
+            os.remove(file_path)
+
+
+def hpss_put(hpss, file, keep=True):
+    """
+    Put a file to the HPSS archive.
+    """
+    hpss_transfer(hpss, file, True, keep)
+
+
+def hpss_get(hpss, file):
+    """
+    Get a file from the HPSS archive.
+    """
+    hpss_transfer(hpss, file, False)
+
+
+def hpss_chgrp(hpss, group, recurse=False):
```

Comment:
Don't remember where this is called from on top of my head. Do we need to do anything here?

Reply:
It's used in chgrp.py (https://github.com/E3SM-Project/zstash/blob/master/zstash/chgrp.py#L23). Yeah I should probably add the "none" check to this function too.